src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7438,7 +7444,7 @@ void Lowering::ContainCheckCompare(GenTreeOp* cmp)
             // One of op1 or op2 could be marked as reg optional
             // to indicate that codegen can still generate code
             // if one of them is on stack.
-            GenTree* regOptionalCandidate = op1->IsCnsIntOrI() ? op2 : PreferredRegOptionalOperand(cmp);
+            GenTree* regOptionalCandidate = op1->IsCnsIntOrI() ? op2 : PreferredRegOptionalOperand(op1, op2);
```

Comment:
This is loosely related, but it seems like we're missing a potential opt here if `op1->IsCnsIntOrI` because we don't do a `CheckImmedAndMakeContained(cmp, op1))` even though this is a commutative scenario and we have that check here so it can presumably be hit.